src/modules/claims/claims.service.ts

```diff
@@ -1448,7 +1448,7 @@ export class ClaimsService {
     const role = vc.credentialSubject.role.namespace;
     let issuerVerified = true;
     try {
-      await this._issuerVerification.verifyIssuer(issuerDID, role);
+      await this._issuerVerification.verifyIssuer(issuerDID as string, role);
```

Comment:
Rather than use `as string`, I think it's better to add a check after `const issuerDID = vc.issuer` so that error can be thrown closer to the source.

Reply:
@jrhender Should we throw an error if no issuer, or just add to the array of errors that there is no issuer and carry on with verification (and not call verify issuer)

Reply:
I guess I would add to the error array and return. If there is no issuer then doing any verification doesn't make sense as it is a required property of the VC data model https://www.w3.org/TR/vc-data-model/#issuer

Reply:
Oh okay - then it that case maybe it would make sense to throw so we can stop the process?

Reply:
Right, if that makes most sense to you, I could see that working. Both ways make sense to me

Reply:
Can the cast be removed with the above check? I would think that it could be
```suggestion
      await this._issuerVerification.verifyIssuer(issuerDID, role);
```

Reply:
How about this change? I think it is better than the cast as before the code would fail if the type was indeed `IIssuer`
```suggestion
      if (typeof issuerDID === 'string') {
        await this._issuerVerification.verifyIssuer(issuerDID, role);
      }
      else {
        await this._issuerVerification.verifyIssuer(issuerDID.id, role);
      }
```